site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/HapiFhirRepository.java

```diff
@@ -80,29 +82,43 @@ public String saveOrganization(final Organization organization) {
     }
 
     /**
-     * Exact search on name
+     * Search by some criteria
      *
-     * @param name of the organization to search.
-     * @return org with that name or none() if not found
+     * @param criterion search criteria
+     * @return stream of orgs matching search
      */
-    @Override
-    public Optional<Organization> findOrganizationByName(final String name) {
+    Stream<Organization> findOrganizations(final ICriterion<?> criterion) {
```

Comment:
findOrganizations is the internal method, not for client use; it returns a stream because you might want to do different things with it - e.g. test for the existence of any org, or filter it, or get the first match, or collect it to some other container, which is what findOrganizationsByPartOf does - clients hardly ever want a stream and would just have to collect it into a list. I'm happy to change it, but that is why they return different types - findOrganizations is part of the internal implementation, but findOrganizationsByPartOf is part of the API.

Reply:
Changed so that the private impl is properly private.